icu4j/maven-build/maven-icu4j/pom.xml

```diff
@@ -0,0 +1,493 @@
+                  </excludes>
+                </resource>
+              </resources>
+            </configuration>
+          </execution>
+
+
+        </executions>
+      </plugin>
+
+      <!--
+        Extract and copy the *.res files from the *.jar files in the upstream depenendency `com.ibm.icu:icu4j-datafiles`.
+        Enables the creation of *.lst files in the next phase using the helper script.
+        Use the ./target/classes directory as the output directory so that the files (included generated *.lst files)
+        get included in the shaded .jar file.
+      -->
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-dependency-plugin</artifactId>
+        <version>3.3.0</version>
+        <executions>
+          <execution>
+            <id>unpack-dependencies</id>
+            <phase>generate-resources</phase>
+            <goals>
+              <goal>unpack-dependencies</goal>
+            </goals>
+            <configuration>
+              <includeGroupIds>com.ibm.icu</includeGroupIds>
+              <!-- exclude test resource files which come through dependencies as `provided` scope -->
+              <excludeScope>provided</excludeScope>
+              <!--
+                Don't know why the Ant build.xml is configured to exclude `**/res_index.res` from the set of
+                included files since those files still appear in the packaged .jar file.
+              -->
+              <excludes>**/*.cnv,**/cnvalias.icu</excludes>
+              <outputDirectory>${project.build.directory}/classes</outputDirectory>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+
+      <!-- 
+        Process and transform the resource files before they get packaged up in the .jar file.
+      -->
+      <plugin>
+          <groupId>org.codehaus.mojo</groupId>
+          <artifactId>exec-maven-plugin</artifactId>
+          <version>3.1.0</version>
+          <executions>
+              <!-- 
+                Replicate the creation of fullLocaleNames.lst files in every subdirectory of the dir of unpacked data
+                that the Ant build.xml file encodes.
+              -->
+              <execution>
+                  <id>full-locale-names</id>
+                  <phase>process-resources</phase>
+                  <goals>
+                      <goal>exec</goal>
+                  </goals>
+                  <configuration>
+                      <executable>./file-listing-helper.sh</executable>
+                  </configuration>
+              </execution>
+              <!-- Copy the license file. -->
+              <execution>
+                  <id>copy-license-file</id>
+                  <phase>process-resources</phase>
+                  <goals>
+                      <goal>exec</goal>
+                  </goals>
+                  <configuration>
+                      <executable>cp</executable>
```

Comment:
OS dependent.
You don't want anything like this in mvn files.

Here are some ways to copy files using plugins:
https://www.baeldung.com/maven-copy-files
And from there I would go with an official maven plugin, like `maven-resources-plugin`

Reply:
See discussion in comment above about needing to take things in a careful step by step approach, including solutions that may be temporary as they are superseded by future improvements.

I couldn't use a plugin because I needed to perform a slightly complex transformation (a creation of certain files per resource file directory, _recursively_) to match the Ant build as a part of the first step requirements.